File: go/src/leetcode/lc0001.go
package leetcode

func twoSum(nums []int, target int) []int {
	positions := make(map[int]int)
	for index, num := range nums {
		position, ok := positions[target-num]
		if ok {
			return []int{position, index}
		}
		positions[num] = index
	}
	return []int{}
}

File: js/leetcode/lc0001.js
var twoSum = function (nums, target) {
    let positions = {};
    for (let index = 0; index < nums.length; index++) {
        let position = positions[target - nums[index]];
        if (position != undefined) return [position, index];
        positions[nums[index]] = index;
    }
    return [];
};

module.exports = {
    twoSum: twoSum
}